# Game Users' Subscription Prediction Using KNN Classification

## Introduction

- <span style="font-size: 15px"> Questions:</span>
The question this project aims

In [4]:
# Load libraries
library(repr)
library(tidyverse)
library(tidymodels)
options(repr.matrix.max.rows = 10)

# Load dataset
players <- read_csv(url("https://raw.githubusercontent.com/linq060119/group-project/refs/heads/main/players.csv"))
slice(players, 1:6)
dim(players)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

[1] 196   7

Players dataset includes 196 observations and 7 variables:

* `experience` (ordinal: Beginner, Amateur, Regular, Veteran, Pro), indicating skill level. 

* `subscribe` (boolean: TRUE/FALSE), reflecting subscription status.

* `hashedEmail` (unique anonymized identifier).

* `played_hours` (float: 0–223.1), includes extreme values 223.1 hours.

* `name` (string; possible duplicates).

* `gender` (categorical: Male, Female, Non-binary, Prefer not to say, Two-spirited, Other).

* `Age` (integer: 8–50; missing values like `NA`), majority of players are teenagers (median = 17).

The data is likely collected from user registrations and activity logs.

Potential issues: `played_hours` includes extreme values 223.1 hours. `Age` includes outlier "50" and `NA`. Logical inconsistencies like some players are labeled as Veteran or Pro with 0 `played_hours`. Too many gender categories.

In this project, we are going to use `played_hours` and `Age` as the quantitative predictors.